File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program94.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept Character from user and check whether it is special symbol or not (!, @, #, $, %, ^, &, *).
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	private:
		char ch;
		
	public:
		Solution(char ch='\0')
		{
			this->ch = ch;
		}

		bool ChkSpecial()
		{
			if(ch>='!' && ch<='/')
			{
				return true;
			}
			else
			{
				return false;
			}


		}


};

int main()
{
	char ch='\0';
	bool bRet=false;

	cout<<"Enter the character: ";
	cin>>ch;

	Solution *dObj = new Solution(ch);
	bRet = dObj->ChkSpecial();

	if(bRet==true)
	{
		cout<<"It is Special symbol\n";
	}
	else
	{
		cout<<"It is not Special symbol\n";
	}

	delete dObj;
	return 0;
}

/*
OUTPUT:
Enter the character: %
It is Special symbol

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program94.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept Character from user and check whether it is special symbol or not (!, @, #, $, %, ^, &, *).
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{

	public char ch;

	public Solution(char ch)
	{
		this.ch = ch;
	}

	public boolean ChkSpecial()
	{	
		
		if(ch>='!' && ch<='/')
		{
			return true;
		}
		else
		{
			return false;
		}
	}


}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);
		boolean bRet = false;

		System.out.println("Enter character:");
		char ch = sObj.next().charAt(0);

		Solution obj = new Solution(ch);
		bRet = obj.ChkSpecial();

		if(bRet == true)
		{
			System.out.println("It is Special Symbol");
		}
		else
		{
			System.out.println("It is not Special Symbol");
		}
	}
}

/*
OUTPUT:
Enter character:
%
It is Special Symbol

*/
